src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
-}
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 
+  static delleteDeadAnimals() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
+}
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (
+      target instanceof Carnivore ||
+      (target instanceof Herbivore && target.hidden)
```

Comment:
The condition to check if the target is a Herbivore and not hidden can be simplified. Since Carnivores should not be able to bite other Carnivores or hidden Herbivores, you can use a single condition to check if the target is an instance of Herbivore and not hidden.